pkg/apis/autoscaling/validation/validation_test.go

```diff
@@ -96,6 +99,10 @@ func TestValidateHorizontalPodAutoscaler(t *testing.T) {
 	if err != nil {
 		t.Errorf("unable to parse label selector: %v", err)
 	}
+
+	// Enable HPAScaleToZero feature gate.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.HPAScaleToZero, true)()
```

Comment:
rather than enabling unconditionally for all tests, we need tests exercising behavior with the feature enabled and disabled

Reply:
I moved the gate switching to separate test functions.